relay-common/src/serde.rs

```diff
@@ -0,0 +1,71 @@
+use serde::ser::Serializer;
+use serde::{Deserialize, Serialize};
+
+/// A wrapper type that is able to deserialize its inner value from a nested JSON string.
+///
+/// For example, this struct:
+///
+/// ```rust
+/// # use relay_common::JsonStringifiedValue;
+/// use serde::Deserialize;
+///
+/// #[derive(Deserialize)]
+/// struct TraceContext {
+///     sample_rate: JsonStringifiedValue<f64>,
+/// }
+/// ```
+///
+/// ...deserializes from `{"sample_rate": "1.0"}` and `{"sample_rate": 1.0}`, and serializes to
+/// `{"sample_rate": "1.0"}`
+///
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
+pub struct JsonStringifiedValue<T>(pub T);
```

Comment:
This type is rather complex for the single use, and allows direct deserialization which we do not actually want in the sampling context. Instead, it would be great if we just use a custom deserialize function or even parse this lazily in an accessor function.